scripts/api-compatibility-check/README.md

```diff
@@ -0,0 +1,83 @@
+# API Compatibility check
+
+## What is it
+
+API compatibility checks help us to ensure that we would not accidentally break a public API. The set of scripts runs Apple's tool `swift-api-digester` to dump public API to a reusable JSON file and compare each other.
+
+## How to run
+
+You can find two scripts in this folder. The first, `breaking-api-check.sh`, designed to run set of public API checks over the pre-built baseline. The second, `rebuild-baseline.sh`, helps to generate new baseline based on git reference. Let's take a closer look on each of them.
+
+### `breaking-api-check.sh`
+
+The `breaking-api-check.sh` accept optional argument `-p path` which should be packaged version of MapboxMaps. Without `-p` argument, script would trigger `scripts/release/packager/package-mapbox-maps.sh` to build XCFramework from scratch. It is useful to path pre-built binary to avoid extra building steps. Breaking API checks running against `.baseline.zip` JSON API digester dump. To update the baseline, you have to proceed to `rebuild-baseline.sh`.
+The script is path-independent so you can call it from any other folder with an absolute or relative path.
+
+### `rebuild-baseline.sh`
+
+The `rebuild-baseline.sh` script simplifies baseline data updates. Usually, we should not update the baseline for the major version. However, my investigations show that digester would trigger breaking changes for `@_spi` changes as well which is not the desired behavior.
+To update the baseline, update the `.baseapi` file to include a new git reference, then call the script. The new git worktree would be checkout in a hidden folder for script purposes. 
+In a successful result, you would get an updated version of `.baseline.zip`. This archive represents folder structure with JSON SDK API dump made by swift-api-digester. It is a native structure for `swift-api-digester` so we can easily extend our platform validations to include macosx/watchos or any other Apple platform.
```

Comment:
```suggestion
`rebuild-baseline.sh` simplifies baseline data updates. Usually, we should not update the baseline within a single major version. However, the digester tool triggers breaking changes for `@_spi` changes as well which is not the desired behavior since we use these to denote experimental APIs which we reserve the right to change outside of a major version bump.

To update the baseline, update the `.baseapi` file to include a new git reference, then call the script. The new git worktree is checked out in a hidden folder for script purposes.

Upon success, you will get an updated version of `.baseline.zip`. This archive represents folder structure with JSON SDK API dump made by swift-api-digester. It is a native structure for `swift-api-digester` so we can easily extend our platform validations to include macosx/watchos or any other Apple platform.
```

> This archive represents folder structure with JSON SDK API dump made by swift-api-digester.

I don't understand what this means.

Reply:
swift-api-digester can generate a baseline folder structure. It looks like: 
Root folder – API – { iphoneos.json |  macosx.json | … }
I am preserving the same folders and files to make them compatible with swift-api-digester.

Reply:
That makes more sense.